### PR TITLE
Fix/multiple ephemeral delete

### LINKED ISSUE #12401: Deleting multiple `ephemeral` instances only removes the first
How to reproduce:

```
$ lxc launch --ephemeral ubuntu:22.04 foo1
Creating foo1
Starting foo1
$ lxc launch --ephemeral ubuntu:22.04 foo2
Creating foo2
Starting foo2
$ lxc list foo
+------+---------+---------------------+-----------------------------------------------+-----------------------+-----------+
| NAME |  STATE  |        IPV4         |                     IPV6                      |         TYPE          | SNAPSHOTS |
+------+---------+---------------------+-----------------------------------------------+-----------------------+-----------+
| foo1 | RUNNING | 10.76.66.80 (eth0)  | fd42:b2a9:e8fd:1156:216:3eff:fe54:e684 (eth0) | CONTAINER (EPHEMERAL) | 1         |
+------+---------+---------------------+-----------------------------------------------+-----------------------+-----------+
| foo2 | RUNNING | 10.76.66.125 (eth0) | fd42:b2a9:e8fd:1156:216:3eff:feb7:cb8d (eth0) | CONTAINER (EPHEMERAL) | 1         |
+------+---------+---------------------+-----------------------------------------------+-----------------------+-----------+

$ lxc delete -f foo1 foo2
$ echo $?
0

$ lxc list foo
+------+---------+---------------------+-----------------------------------------------+-----------------------+-----------+
| NAME |  STATE  |        IPV4         |                     IPV6                      |         TYPE          | SNAPSHOTS |
+------+---------+---------------------+-----------------------------------------------+-----------------------+-----------+
| foo2 | RUNNING | 10.76.66.125 (eth0) | fd42:b2a9:e8fd:1156:216:3eff:feb7:cb8d (eth0) | CONTAINER (EPHEMERAL) | 1         |
+------+---------+---------------------+-----------------------------------------------+-----------------------+-----------+
```

**Note**: only ephemeral instances are affected.

Version information:

```
$ snap list lxd
Name  Version      Rev    Tracking     Publisher   Notes
lxd   git-0eed2a3  26051  latest/edge  canonical✓  -
```

### PR DESCRIPTION
**Pull request by gabrielmougard**
closes #12401

### CODE REVIEW

File: test/suites/basic.sh [hunk 1]
```diff
@@ -655,4 +655,14 @@ test_basic_usage() {
...
+
+  # Test deleting multiple ephemeral containers
+  lxc launch --ephemeral testimage c1
+  lxc launch --ephemeral testimage c2
+  lxc launch --ephemeral testimage c3
+
+  lxc delete -f c1 c2 c3
+  # check that the containers are really gone
+  remaining_instances=$(lxc list --format --csv | wc -l)
+  [ "${remaining_instances}" -eq 0 ]
```
Comment (simondeziel) [inline test/suites/basic.sh:667]:
```
  remaining_instances="$(lxc list --format csv)"
  [ -z "${remaining_instances}" ]
```

File: lxc/delete.go [hunk 2]
```diff
@@ -134,8 +135,13 @@ func (c *cmdDelete) Run(cmd *cobra.Command, args []string) error {
 				return fmt.Errorf(i18n.G("Stopping the instance failed: %s"), err)
 			}
 
+			resourceCount -= 1
 			if ct.Ephemeral {
-				return nil
+				if resourceCount == 0 {
```
Comment (tomponline) [inline lxc/delete.go:140]: Can we just continue here , why do we need the resource count logic?

File: lxc/delete.go [hunk 3]
```diff
@@ -135,7 +135,11 @@ func (c *cmdDelete) Run(cmd *cobra.Command, args []string) error {
 			}
 
 			if ct.Ephemeral {
-				return nil
+				if i == len(resources)-1 {
```
Comment (tomponline) [inline lxc/delete.go:138]: just continue will suffice, why do we need the logic wrt to resources?

Comment (gabrielmougard) [reply lxc/delete.go:138]: We don't want to call the doDelete function if the instance is ephemeral right ? That's why we return early

Comment (gabrielmougard) [reply lxc/delete.go]: @tomponline it should be good now

**Review by tomponline — APPROVED** [review]
Thats more like it :)

**Pull request merged**